src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,325 @@
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set start time of a guild scheduled event
+   * guildScheduledEvent.setScheduledStartTime('September 24, 2022'))
+   *  .then(guildScheduledEvent => console.log(`Set the start time to: ${guildScheduledEvent.scheduledStartTime}`))
+   *  .catch(console.error);
+   */
+  setScheduledStartTime(scheduledStartTime) {
+    return this.edit({ scheduledStartTime });
+  }
+
+  /**
+   * Sets a new time to end the event at.
+   * @param {?DateResolvable} scheduledEndTime The time to end the event at
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set end time of a guild scheduled event
+   * guildScheduledEvent.setScheduledEndTime('September 25, 2022'))
+   *  .then(guildScheduledEvent => console.log(`Set the end time to: ${guildScheduledEvent.scheduledEndTime}`))
+   *  .catch(console.error);
+   */
+  setScheduledEndTime(scheduledEndTime) {
+    return this.edit({ scheduledEndTime });
+  }
+
+  /**
+   * Sets the new description of the guild scheduled event.
+   * @param {string} description The description of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set description of a guild scheduled event
+   * guildScheduledEvent.setDescription('A virtual birthday party')
+   *  .then(guildScheduledEvent => console.log(`Set the description to: ${guildScheduledEvent.description}`))
+   *  .catch(console.error);
+   */
+  setDescription(description) {
+    return this.edit({ description });
+  }
+
+  /**
+   * Sets the new status of the guild scheduled event.
+   * @param {GuildScheduledEventStatus|number} status The status of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set status of a guild scheduled event
+   * guildScheduledEvent.setStatus('ACTIVE')
+   *  .then(guildScheduledEvent => console.log(`Set the status to: ${guildScheduledEvent.status}`))
+   *  .catch(console.error);
+   */
+  setStatus(status) {
+    return this.edit({ status });
+  }
+
+  /**
+   * Sets the new location of the guild scheduled event.
+   * @param {string} location The location of the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   * @example
+   * // Set location of a guild scheduled event
+   * guildScheduledEvent.setLocation('Earth')
+   *  .then(guildScheduledEvent => console.log(`Set the location to: ${guildScheduledEvent.location}`))
+   *  .catch(console.error);
+   */
+  setLocation(location) {
+    return this.edit({ location });
+  }
+
+  /**
+   * Fetches subscribers of this guild scheduled event.
+   * @param {FetchGuildScheduledEventSubscribersOptions} [options] Options for fetching the subscribers
+   * @returns {Promise<Collection<Snowflake, User> | Collection<Snowflake, GuildMember>>}
+   */
+  fetchSubscribers(options) {
+    return this.guild.scheduledEvents.fetchSubscribers(this.id, options);
+  }
```

Comment:
Seeing as there is a property called userCount, shouldn't this be fetchUsers, or possibly even moved into a GuildScheduledEventUserManager? There are events for these so we could cache them pretty easily

Reply:
1) The method name is self-descriptive currently. A subscriber can be a user as well as a member, so it makes sense to keep it as it is.
2) Those two events are experimental currently, so maybe in some other PR??

Reply:
How are the events experimental if they’re included in this PR

Reply:
Only added the basic support for them, adding more features around experimental events doesn't seem right.